minari/dataset/minari_dataset.py

```diff
@@ -349,42 +310,23 @@ def update_dataset_from_buffer(self, buffer: List[dict]):
         Args:
             buffer (list[dict]): list of episode dictionary buffers to add to dataset
         """
-        additional_steps = 0
-        with h5py.File(self.spec.data_path, "a", track_order=True) as file:
-            last_episode_id = file.attrs["total_episodes"]
-            for i, eps_buff in enumerate(buffer):
-                episode_id = last_episode_id + i
-                # check episode terminated or truncated
-                assert (
-                    eps_buff["terminations"][-1] or eps_buff["truncations"][-1]
-                ), "Each episode must be terminated or truncated before adding it to a Minari dataset"
-                assert len(eps_buff["actions"]) + 1 == len(
-                    eps_buff["observations"]
-                ), f"Number of observations {len(eps_buff['observations'])} must have an additional \
-                                                                                        element compared to the number of action steps {len(eps_buff['actions'])} \
-                                                                                        The initial and final observation must be included"
-                seed = eps_buff.pop("seed", None)
-                episode_group = clear_episode_buffer(
-                    eps_buff, file.create_group(f"episode_{episode_id}")
-                )
-
-                episode_group.attrs["id"] = episode_id
-                total_steps = len(eps_buff["actions"])
-                episode_group.attrs["total_steps"] = total_steps
-                additional_steps += total_steps
-
-                if seed is None:
-                    episode_group.attrs["seed"] = str(None)
-                else:
-                    assert isinstance(seed, int)
-                    episode_group.attrs["seed"] = seed
-
-                # TODO: save EpisodeMetadataCallback callback in MinariDataset and update new episode group metadata
-
-            file.attrs.modify("total_episodes", last_episode_id + len(buffer))
-            file.attrs.modify(
-                "total_steps", file.attrs["total_steps"] + additional_steps
-            )
+        old_total_episodes = self._data.total_episodes
+
+        self._data.update_from_buffer(buffer, self.spec.data_path)
+
+        new_total_episodes = self._data._total_episodes
+
+        self._episode_indices = np.append(
+            self._episode_indices, np.arange(old_total_episodes, new_total_episodes)
+        )  # ~= np.append(self._episode_indices,np.arange(self._data.total_episodes))
+
+        self.spec.total_episodes = len(self._episode_indices)
```

Comment:
(a whim): be consistent with init (there you used `.size`)

Reply:
done